app/src/main/java/com/github/se/polyfit/ui/components/AddIngredientPopup.kt

```diff
@@ -0,0 +1,193 @@
+package com.github.se.polyfit.ui.components
+
+import android.util.Log
+import androidx.compose.foundation.*
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import com.github.se.polyfit.ui.theme.*
+import com.github.se.polyfit.ui.utils.removeLeadingZerosAndNonDigits
+
+// Constants
+// Add to list for new nutrition fields
+val NUTRITION_LABELS = listOf("Serving Size", "Calories", "Carbs", "Fat", "Protein")
+val NUTRITION_UNITS = listOf("g", "kcal", "g", "g", "g")
+// Temporary list of available ingredient to search
+val TMP_AVAILABLE_INGREDIENT = listOf("Apple", "Banana", "Carrot", "Date", "Eggplant", "apes")
+
+@Composable
+fun EditIngredientNutrition() {
```

Comment:
Either take an `Ingredient` as a parameter or maybe an `IngredientViewModel`? @RandomUsername1315 I'm not sure the structure we've gone with yet

Reply:
added a TODO